java-squid/src/main/java/org/sonar/java/ast/parser/TypeParameterListTreeImpl.java

```diff
@@ -77,7 +78,17 @@ public void accept(TreeVisitor visitor) {
 
   @Override
   public Iterator<Tree> childrenIterator() {
-    return ImmutableList.<Tree>builder().addAll(this).build().iterator();
+    ImmutableList.Builder<Tree> iterator = ImmutableList.<Tree>builder();
+    if (openBracketToken != null) {
+      iterator.add(openBracketToken);
+    }
+    // FIXME SONARJAVA-547 separators between parameters are currently ignored
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)